### ISSUE TITLE
Cannot connect to Azure SQL Server on ubuntu-18.04 

### ISSUE THREAD

**Issue by AshTappin** [actions/virtual-environments#746]
**Describe the bug**
I was using windows-2019 previously for a pipeline build. It was making successful tcp connections to an Azure SQL Server ("Allow Azure services and resources to access this server" is enabled) However, I had to change the vmImage to ubuntu-18.04 and now it can't make a connection to the SQL server.

This is the area I am getting (the build is running Selenium tests that directly access the database):

`System.Data.SqlClient.SqlException : A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 40 - Could not open a connection to SQL Server)
`
I don't understand why the connection is working with `windows-2019` but not `ubuntu-18.04`?

**Question, Bug, or Feature?**: 
Question

**Virtual environments affected**
- [ ] Ubuntu 18.04 LTS

**Comment by al-cheb** [comment]
Hello, @AshTappin 

Could you please provide results on windows and ubuntu powershell code?

```
$server="sqlserver"
$port=1433

[System.Net.Dns]::Resolve($server)
try {
	(new-object Net.Sockets.TcpClient).Connect($server,$port)
} catch {
	"Failed to connect $server on port $port"
} 
```

**Comment by AshTappin** [comment]
@al-cheb We can close the issue, I figured out what the problem was. 

I ran a netstat command and it was connecting to the Azure SQL Server successfully from the ubuntu VM. The problem was related to environment variables. I am setting a job variable which had the database connection string. I didn't realise when job variables are copied to environment variables, they are converted to upper case (https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch). Since envs in Windows are case insensitive, I wasn't getting a problem when running the build on windows-2019. My tests were looking for environment variable `Foo` rather than `FOO`. Updated to upper case and my build is working on ubuntu-18.04 now :)
